internal/provider/aws/health.go

```diff
@@ -0,0 +1,254 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/aws/aws-sdk-go/service/route53/route53iface"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	externaldns "sigs.k8s.io/external-dns/endpoint"
+
+	"github.com/rs/xid"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/internal/provider"
+)
+
+const (
+	idTag = "kuadrant.dev/healthcheck"
+
+	defaultHealthCheckPath             = "/"
+	defaultHealthCheckPort             = 80
+	defaultHealthCheckFailureThreshold = 3
+)
+
+var (
+	callerReference func(id string) *string
+)
+
+type Route53HealthCheckReconciler struct {
+	client route53iface.Route53API
+}
+
+var _ provider.HealthCheckReconciler = &Route53HealthCheckReconciler{}
+
+func NewRoute53HealthCheckReconciler(client route53iface.Route53API) *Route53HealthCheckReconciler {
+	return &Route53HealthCheckReconciler{
+		client: client,
+	}
+}
+
+func (r *Route53HealthCheckReconciler) Reconcile(ctx context.Context, spec provider.HealthCheckSpec, endpoint *externaldns.Endpoint, probeStatus *v1alpha1.HealthCheckStatusProbe) (provider.HealthCheckResult, error) {
+	healthCheck, exists, err := r.findHealthCheck(ctx, probeStatus)
+	if err != nil {
+		log.Log.Info("error finding health check", "err", err)
```

Comment:
should we log this here or just add it as context to the returned error? I imagine it will get logged here and then back at the dnsrecord reconciler?